bsearch.py

```diff
@@ -1,6 +1,27 @@
-from typing import Iterable
+"""
+implement binary search
+"""
 
 
-def bsearch(l: Iterable[int], v:int) -> int:
-    """l always is sorted"""
-    return 42
\ No newline at end of file
+def bin_search(lst, num: int) -> int:
+    """
+
+    :param lst:
+    :param num:
+    :return:
+    """
+    if len(lst) == 0 or num == 0:
+        raise ValueError
+    first = 0
+    last = len(lst) - 1
+    mid = 0
+    while first <= last:
+        mid = (first + last) // 2
+        if lst[mid] != num:
+            if num < lst[mid]:
+                last = mid - 1
+            else:
+                first = mid + 1
+        elif lst[mid] == num:
```

Comment:
Use else here, you have only two cases. Also you can rewrite this in such way:
if lst[mid] == num:
   break
if num < lst[mid]:
                last = mid - 1
            else:
                first = mid + 1